### PR TITLE
Phase 4: TiDE model-family schema rollout + checklist cleanup

### PR DESCRIPTION
**Pull request by RobotPsychologist**
## Summary

This PR rolls TiDE into the schema-routed model family workflow and performs TiDE model/config cleanup aligned to the model implementation quality checklist.

### What changed

- Added `TiDEModelConfigSchema` and TiDE runtime adapter route in `src/config/schemas/model_configs.py`.
- Exported TiDE schema/adapter symbols from `src/config/schemas/__init__.py`.
- Added TiDE schema artifact generation in `src/config/schemas/json_schema_artifacts.py`.
- Wired `ModelFactory.create_model` TiDE path through `build_model_runtime_config(...)` in `src/workflows/forecasting/modeling.py`.
- Added TiDE-specific training mode default handling in `ModelFactory.create_finetune_config` (`from_scratch` by default for TiDE).
- Tightened TiDE config invariants in `src/models/tide/config.py`:
  - enforce `training_mode == "from_scratch"`
  - enforce `scaling == "mean"`
  - enforce `encoder_hidden_dim == decoder_hidden_dim`
  - canonicalize/sync `learning_rate` and `lr` semantics
- Refactored duplicated TiDE inference prep/quantile extraction code in `src/models/tide/model.py` into private helpers.
- Expanded schema/factory tests in `tests/workflows/forecasting/test_model_config_schema_loader.py` for TiDE:
  - fixture-backed schema loading
  - alias normalization
  - architectural constraint failures
  - default-parity checks
  - dedicated factory-path coverage
  - unknown-field fail-fast behavior
- Regenerated schema artifacts and manifest, including `model_configs.tide.schema.json`.
- Updated rollout tracking and docs pages.

---

### 1) Public model interface remains consistent with shared contract
**Addressed:** TiDE still uses typed `TiDEConfig`, remains a `BaseTimeSeriesFoundationModel` implementation, and keeps required core properties/methods (`training_backend`, `supports_zero_shot`, probabilistic support, predict/load/save hooks).

### 2) Any required deviations are explicitly documented and justified
**Addressed:** TiDE-specific constraints are explicit and enforced (`from_scratch` only, `scaling='mean'`, encoder/decoder dim parity), matching backend/model requirements.

### 3) Dead/duplicate code removed from touched surfaces
**Addressed:** Duplicated prediction-context and quantile extraction logic in TiDE model inference paths was consolidated into private helpers.

### 4) Module/class/method ordering follows repository standard
**Addressed:** TiDE modules retain module docstring → imports → class definitions; helper methods are private and placed after public lifecycle/inference/persistence sections.

### 5) Docstrings updated to reflect actual behavior
**Addressed:** TiDE config docstring now includes enforced constraints; helper docstrings added for extracted inference utilities.

### 6) Touched files pass lint with zero warnings/errors
**Addressed:** `ruff check` passed for all touched Python files.

### 7) Required family test contract is present and passing
**Addressed:** Added TiDE schema+factory tests in `test_model_config_schema_loader.py` covering:
- valid fixture load,
- invalid constraints,
- alias behavior,
- default parity,
- dedicated factory-path wiring,
- unknown-field rejection.

### 8) Config schema artifacts regenerated when schema contracts changed
**Addressed:** Regenerated `docs/architecture/generated-config-schemas/*`, including new `model_configs.tide.schema.json` and updated `manifest.json`.

### 9) Model PR description links checklist and states validated gates
**Addressed:** This section is the explicit gate-by-gate checklist validation for this model PR.

---

## Validation run

- `ruff check src/config/schemas/model_configs.py src/config/schemas/__init__.py src/config/schemas/json_schema_artifacts.py src/workflows/forecasting/modeling.py src/models/tide/config.py src/models/tide/model.py tests/workflows/forecasting/test_model_config_schema_loader.py`
- `pytest -q tests/workflows/forecasting/test_model_config_schema_loader.py tests/workflows/forecasting/test_config_schema_artifact_generation.py tests/models/test_model_family_contract_suite.py`

Result: **51 passed**.

### CODE REVIEW

**Comment by RobotPsychologist** [comment]
Follow-up fix pushed in `7a6c837` for review feedback on TiDE LR and import linting.

### 1) Why `0.000931` / `9.31e-4` exists
This value was already the TiDE family baseline in existing model configs (`configs/models/tide/*.yaml`) and in `TiDEConfig` defaults before the schema rollout. It is intentionally preserved as the TiDE default learning rate.

### 2) Why line-94 style check existed
The previous logic was trying to reconcile base `ModelConfig.learning_rate` with TiDE-specific `lr`, but the implementation used a hardcoded sentinel (`9.31e-4`), which is brittle and opaque.

### 3) What was changed now
- Removed hardcoded reconciliation behavior.
- Set both TiDE defaults explicitly to `9.31e-4` (`learning_rate` and `lr`) so defaults are aligned.
- Replaced sentinel logic with default-aware reconciliation:
  - if only one of `learning_rate` or `lr` is overridden, mirror it,
  - if both are explicitly set and conflict, raise a clear `ValueError`.

### 4) Why import lint issues were being missed
You were right: our previous `ruff check` path did not enforce import-order linting. Ruff default ruleset does not include `I`.

### 5) Lint setup hardening applied
- Added to repo config (`pyproject.toml`):
  - `[tool.ruff.lint]`
  - `extend-select = ["I"]`
- This makes normal `ruff check` catch import-order issues by default on touched files.
- Applied import-order fixes in touched files on this branch.

### Validation rerun
- `ruff check src/models/tide/config.py src/models/tide/model.py src/config/schemas/model_configs.py tests/workflows/forecasting/test_model_config_schema_loader.py pyproject.toml`
- `pytest -q tests/workflows/forecasting/test_model_config_schema_loader.py tests/workflows/forecasting/test_config_schema_artifact_generation.py tests/models/test_model_family_contract_suite.py`
- Result: **51 passed**.

**Comment by RobotPsychologist** [comment]
Addressed requested follow-ups in commit `0af6458`.

### 1) TiDE default LR rounded
- Updated TiDE defaults from `9.31e-4` to **`1.0e-3`** in:
  - `src/models/tide/config.py` (`learning_rate`, `lr`)
  - `src/config/schemas/model_configs.py` (`TiDEModelConfigSchema.lr`)
- Regenerated schema artifact:
  - `docs/architecture/generated-config-schemas/model_configs.tide.schema.json`
- Updated default-parity test expectation accordingly.

### 2) Local import pattern (`src.` -> relative)
Converted local imports in the TiDE rollout surfaces to package-relative imports (no `from src...` in these modules anymore):
- `src/models/tide/config.py`
- `src/models/tide/model.py`
- `src/workflows/forecasting/modeling.py`

### 3) pyright excludes cleanup
Removed all three requested excludes from `pyrightconfig.json`:
- `**/models/ttm`
- `**/models/ARIMA`
- `**/mlflow_experiments`

### Validation
- `ruff check src/models/tide/config.py src/models/tide/model.py src/workflows/forecasting/modeling.py src/config/schemas/model_configs.py tests/workflows/forecasting/test_model_config_schema_loader.py`
- `pytest -q tests/workflows/forecasting/test_model_config_schema_loader.py tests/workflows/forecasting/test_config_schema_artifact_generation.py tests/models/test_model_family_contract_suite.py`
- Result: **51 passed**.
- VS Code Problems check on touched TiDE/modeling files reports no errors.

**Review by RobotPsychologist — COMMENTED** [review]
LGTM
